src/choleskyQR.jl

```diff
@@ -171,23 +216,72 @@ function resizedata!(K::QRJacobiBand, nm::Integer)
     end
     K
 end
-function cache_filldata!(J::QRJacobiBand{:dv,T}, inds::UnitRange{Int}) where T
+function cache_filldata!(J::QRJacobiBand{:dv,:Q,T}, inds::UnitRange{Int}) where T
+    b = 1+bandwidths(J.U.factors)[2]÷2
+    # pre-fill cached arrays to avoid excessive cost from expansion in loop
+    m, jj = inds[end], inds[2:end]
+    X = jacobimatrix(J.P)[1:m+b+2,1:m+b+2]
+    getindex(J.U.factors,m+b,m+b)
+    getindex(J.U.τ,m)
+    M = J.UX
+    τ = J.U.τ
+    F = J.U.factors
+    dv = J.data
+    @inbounds for n in jj
+        v = [zero(T);one(T);F[n-1:n+b,n][3:end]]
+        H = I-τ[n]*v*v'
+        K = H*M*H
+        M = Matrix(X[n:n+b+1,n:n+b+1])
```

Comment:
Pre-allocate `M` here. It's not a good idea to be allocating memory inside a loop